src/Block/SharedBlockBlockService.php

```diff
@@ -126,14 +136,14 @@ public function preUpdate(BlockInterface $block): void
         $block->setSetting('blockId', \is_object($block->getSetting('blockId')) ? $block->getSetting('blockId')->getId() : null);
     }
 
-    protected function getBlockBuilder(FormMapper $form): FormBuilderInterface
+    protected function getBlockBuilder(): FormBuilderInterface
     {
         $fieldDescription = $this->sharedBlockAdmin->createFieldDescription('block', [
             'translation_domain' => 'SonataPageBundle',
             'edit' => 'list',
         ]);
 
-        return $form->create('blockId', ModelListType::class, [
+        return $this->sharedBlockAdmin->getFormBuilder()->create('blockId', ModelListType::class, [
```

Comment:
createMethod is deprecated, this is how we did for media bundle: https://github.com/sonata-project/SonataMediaBundle/blob/4.x/src/Block/FeatureMediaBlockService.php#L232